[3.5] bpo-32563: Get expat to compile under C89

ncoghlan opened the pull request:
expat: Add artificial scopes in xmltok.c utf8_toUtf8() to fix c89 compilation.

Cherry-picked from libexpat commit e0b290eb3d8f4c4b45137a7d7f4f8db812145bd2

<!-- issue-number: bpo-32563 -->
https://bugs.python.org/issue32563
<!-- /issue-number -->

ncoghlan committed:
[3.5] bpo-32563: Get expat to compile under C89

expat: Add artificial scopes in xmltok.c utf8_toUtf8() to fix c89 compilation.

Cherry-picked from libexpat commit e0b290eb3d8f4c4b45137a7d7f4f8db812145bd2

tiran reviewed (approved):
LGTM

It's not pretty but as long as it works...

vstinner reviewed (commented):
Please mention in your commit message that it's a cherry-pick from upstream. Extract of my 3.4 fix:

```
expat: Add artificial scopes in xmltok.c utf8_toUtf8() to fix c89 compilation.

Cherry-picked from libexpat commit e0b290eb3d8f4c4b45137a7d7f4f8db812145bd2
```

larryhastings committed:
Merge branch '3.5' into bpo-32563-fix-py35-build-error

ncoghlan committed:
Backport XML RPC test skip to 3.5

The buildbot service upgrade removed the XML-RPC
interface, so this test no longer works (through no
fault of the standard library).

ncoghlan committed:
Don't configure a nonexistent Python

larryhastings committed:
Merge branch '3.5' into bpo-32563-fix-py35-build-error
